### PR TITLE
7635 Added "historical" handling + UI and other cleanup

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#7635

*Description of changes:*

Commit 1:
- app version: 4.0.15
- removed HTML base CSS that did nothing
- added some base classes
- added Historical chip/block to EntityInfo
- updated email/phone behaviour/styles in EntityInfo
- removed component classes already defined in base
- combined common filing templates in FHL
- added code for dissolution filings in FHL
- updated/simplified layouts in Todo List
- updated/simplified FHL detail component layouts
- added CompletedDissolution FHL detail component
- cleaned up Todo List detail components
- updated ContactInfo layout in dialogs
- updated DissolutionTypes enum
- updated DateMixin to display long month names
- hide Addresses and Directors change buttons if historical
- added CompletedDissolution unit tests
- updated misc unit tests

Commits 2, 3, 4, 5, 6:
- added prototype code - we don't yet know what the data or enums will look like
- feature is disabled by hard-coded value in getters
- added/updated unit tests
- changed future effective IA fallback text from "this Numbered Benefit Company" to "this company"
- updated whitespace, font size and colour in ContactInfo
- generalized font sizes (variables and classnames)
- misc cleanup
- imported latest StaffComponent (without obsolete color)
- moved staff comments button up to line up with rest of line

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/assets/styles/base.scss
Comment (severinbeauvais) [inline src/assets/styles/base.scss:37]: The 'html' block is only used in the index.html file... after that everything is in 'body'. So this CSS did nothing (decoy pattern).

File: src/assets/styles/base.scss
Comment (severinbeauvais) [inline src/assets/styles/base.scss:153]: This was not used. And actually, there's a Vuetify helper class exactly for this.

File: src/assets/styles/theme.scss
Comment (severinbeauvais) [inline src/assets/styles/theme.scss:30]: I think StaffComment should actually use `$app-blue`, but that's not in scope for this ticket.

File: src/components/Dashboard/FilingHistoryList.vue
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:904]: I moved this to the v-btn with class `menu-btn`. It seems to make the elements more consistent in the template.

File: src/components/Dashboard/FilingHistoryList/CompletedAlteration.vue
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/CompletedAlteration.vue:13]: All paragraphs in all detail components have top and bottom margin as 1rem. This is the exception, since we want both lines to have no extra whitespace between them.

File: src/components/Dashboard/FilingHistoryList/CompletedIa.vue
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/CompletedIa.vue:9]: Using Vuetify helpers to simplify things.
I left the classname for validation purposes.

File: src/components/Dashboard/FilingHistoryList/CompletedIa.vue
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/CompletedIa.vue:37]: H4 style is now in base.scss.

File: src/components/Dashboard/FilingHistoryList/CompletedIa.vue
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/CompletedIa.vue:34]: All detail component paragraphs now have top and bottom margin as 1rem.

Note: margins are used for spacing outside the element (where padding is for spacing inside the element). In this case, we want whitespace  between elements, so margins make sense. Also, margins can overlap, which simplifies things.

File: src/components/Dashboard/FilingHistoryList/FutureEffective.vue
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/FutureEffective.vue:20]: I made this consistent in all detail components. I tested a few of these and they look good.

File: src/components/Dashboard/FilingHistoryList/PendingFiling.vue
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/PendingFiling.vue:16]: The top padding was previously 1 rem. Now the top margin is 1.5rem. This is in line with the latest UI designs.

File: src/components/Dashboard/TodoList.vue
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:32]: This layout change (and on line 41) was needed to get rid of weird special case whitespaces in sub-components. This is way cleaner!

File: tests/unit/EntityInfo.spec.ts
Comment (severinbeauvais) [inline tests/unit/EntityInfo.spec.ts:413]: This is more exact.

File: src/views/Dashboard.vue
Comment (severinbeauvais) [inline src/views/Dashboard.vue:53]: This will be ticket /bcgov/entity#8822.

File: src/mixins/date-mixin.ts
Comment (severinbeauvais) [inline src/mixins/date-mixin.ts:78]: This should be ported to the other UIs. Any takers for Create UI (and possibly BCRS Shared Components)?

File: src/interfaces/history-item-interface.ts
Comment (severinbeauvais) [inline src/interfaces/history-item-interface.ts:55]: I debated whether to just have `isComplete`, `isFutureEffective` and `isFutureEffectivePending` and then force the template to check the entity type, but that's actually much messier in the template (plus, the template doesn't have to re-compute all the conditions when it re-displays due to reactivity). So, this is actually cleaner IMHO.

Also, note that these states are now consistent between IAs/Alterations/Dissolutions.

File: src/components/EntityInfo.vue
Comment (severinbeauvais) [inline src/components/EntityInfo.vue:83]: I added `pa-1` so the SVG icon is better centered in this block.

File: src/components/EntityInfo.vue
Comment (severinbeauvais) [inline src/components/EntityInfo.vue:57]: I tweaked how the different buttons get spaced out. This is now more consistent / no special cases.

File: src/components/EntityInfo.vue
Comment (severinbeauvais) [inline src/components/EntityInfo.vue:137]: Note that disabling pointer events means the click handler won't get called.

File: src/assets/styles/base.scss
Comment (cameron-freshworks) [inline src/assets/styles/base.scss:63]: Unique sizing!

File: src/assets/styles/base.scss
Comment (cameron-freshworks) [inline src/assets/styles/base.scss:256]: Even more unique styling! 
I imagine design calls for these.. but i am curious why so specific

File: src/components/Dashboard/FilingHistoryList.vue
Comment (cameron-freshworks) [inline src/components/Dashboard/FilingHistoryList.vue:252]: O nice, i've not seen one yet. Was there some test data loaded in for this?

File: src/components/EntityInfo.vue
Comment (cameron-freshworks) [reply src/components/EntityInfo.vue:57]: Sounds good! Just to double check the icons still still align with the headers above? *see pic
Tracey was particular about his in a recent review

Screen Shot 2021-11-05 at 8 05 44 AM
.

**Review by cameron-freshworks — APPROVED** [review]
Looks good

File: src/assets/styles/base.scss
Comment (severinbeauvais) [reply src/assets/styles/base.scss:63]: This was existing actually - used by View and Change Company Information button and Dissolve This Company button. I just made it more explicit and global.

File: src/assets/styles/base.scss
Comment (severinbeauvais) [reply src/assets/styles/base.scss:256]: I didn't choose this font; I only commented it.

It's used in the dialogs. I've asked Tracey if we can change this to 14px or 15px.

File: src/components/Dashboard/FilingHistoryList.vue
Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList.vue:252]: Thor said he would create some. I have followed up with him.
For this, I temporarily hacked the UI code to display whatever detail component I wanted.

File: src/components/EntityInfo.vue
Comment (severinbeauvais) [reply src/components/EntityInfo.vue:57]: LGTM:

image

File: src/components/EntityInfo.vue
Comment (severinbeauvais) [reply src/components/EntityInfo.vue:137]: Yes, that's the point :)  I don't want the click event to fire when the entity is historical. (I also added a safety check in the handler itself.)

File: src/App.vue
Comment (severinbeauvais) [inline src/App.vue:500]: I'm very tempted to merge in this code as-is. The functionality is disabled in the `isHistorical` getter anyway (which needs to be wired up however we get data from the API). This is so my code based doesn't fall behind. Thoughts?

File: src/components/Dashboard/FilingHistoryList/FutureEffective.vue
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/FutureEffective.vue:47]: I changed this because we will soon support CP IA, in other words, `isFutureEffectiveIa` could be true, but the company label would be wrong. The company label is now coded (in the HTML) as the entity name (if if exists) or "this company".

Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList/FutureEffective.vue:47]: Oops, I need to change this in FutureEffectivePending.vue as well... next commit.

File: src/components/common/ContactInfo.vue
Comment (severinbeauvais) [inline src/components/common/ContactInfo.vue:5]: A couple of small changes requested by Tracey: more whitespace before the text here, and some font tweaks below.

Comment (severinbeauvais) [reply src/components/common/ContactInfo.vue:5]: (I tested in detail components and a couple of dialogs and it looks good.)

Comment (severinbeauvais) [reply src/components/common/ContactInfo.vue:5]: In a detail component:

image

Comment (severinbeauvais) [reply src/components/common/ContactInfo.vue:5]: In a dialog:

image

File: src/enums/dissolutionTypesNames.ts
Comment (severinbeauvais) [inline src/enums/dissolutionTypesNames.ts:1]: This depends on what values the API uses. The sample data I have is from COLIN, I think, and I suspect LEAR will only use a subset of these.

In the meantime, merging this code probably (*) won't hurt anything.

(*) The Create UI will need the same enums. Actually I was thinking about moving these to BCRS Shared Components except these aren't finalized yet.

File: src/mixins/filing-mixin.ts
Comment (severinbeauvais) [inline src/mixins/filing-mixin.ts:135]: I'm not 100% sure what Create UI will do with this draft data atm. We need to work this through...

File: tests/unit/FutureEffective.spec.ts
Comment (severinbeauvais) [inline tests/unit/FutureEffective.spec.ts:1]: You'll notice I verify the entire lines of text below. I can't think of a reason not to.

I formatted them as below for easier readability.

File: src/assets/styles/base.scss
Comment (severinbeauvais) [inline src/assets/styles/base.scss:33]: This class can be used anywhere in this app, but it can only be `@extend`ed where it's visible, ie, in this file only (since no components import base.scss).
So, I also created font size variables that are used in the components - see theme.scss.

Comment (severinbeauvais) [reply src/assets/styles/base.scss:33]: What I mean is, some components make use of this class, while other components use the variables instead.

File: src/components/Dashboard/FilingHistoryList/FutureEffectivePending.vue
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/FutureEffectivePending.vue:43]: `ifFutureEffectiveIaPending` can be True for Coop IAs, so I changed "this Numbered Benefit Company" to "this company" and moved it into the template (same as FutureEffective.vue).

File: src/components/Dashboard/AddressListSm.vue
Comment (severinbeauvais) [inline src/components/Dashboard/AddressListSm.vue:252]: This isn't an exact replacement, but the purpose of this font size is to be smaller than 15, which this is. (No one can tell the different between 0.85rem and 0.875rem.)

I also did this in a couple of other components where the font size was 0.8, 0.85 or 0.9.

File: src/components/common/ContactInfo.vue
Comment (severinbeauvais) [reply src/components/common/ContactInfo.vue:5]: In a dialog:

dialog with 15px body text

^^^ I updated this dialog with the 15px body text (instead 14.4px as before). This was OKed by Tracey.

File: src/mixins/date-mixin.ts
Comment (severinbeauvais) [reply src/mixins/date-mixin.ts:78]: OK, I've got this as an item in my todo list ... I will get to it some time.

File: src/assets/styles/base.scss
Comment (severinbeauvais) [reply src/assets/styles/base.scss]: Guess how many font sizes we use in this app... (rhetorical question)

(Too many. I think Tracey wants us to look at this some time in the future.)

File: src/assets/styles/base.scss
Comment (severinbeauvais) [reply src/assets/styles/base.scss]: Tracey is OK with 15px. I have made the change.

Comment (severinbeauvais) [reply src/assets/styles/base.scss]: (14 looked too small and 16 looked too large. 15 is the Goldilocks number.)

File: src/App.vue
Comment (severinbeauvais) [reply src/App.vue:500]: @argush3 Any thoughts on this? Linda was OK with it at s/u this morning. I'm just waiting for Cam's and your final reviews.

Comment (argush3) [reply src/App.vue:500]: yup, i'm good with merging the code into main so it doesn't go stale.  will try to get my review in before EOD.

**Review by eve-git — APPROVED** [review]

File: src/assets/styles/base.scss
Comment (cameron-freshworks) [reply src/assets/styles/base.scss]: haha yes, i can't imagine :)

File: src/assets/styles/base.scss
Comment (cameron-freshworks) [reply src/assets/styles/base.scss]: Perfect :)

File: src/components/Dashboard/FilingHistoryList.vue
Comment (cameron-freshworks) [reply src/components/Dashboard/FilingHistoryList.vue:252]: Sounds good!

Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList.vue:252]: Thanks for resolving this conversation. As I mentioned earlier, after this PR is merged I will park this ticket until we find out more what the API will provide. At that time there will be real dissolution filings / entities to look at. No further action needed here atm.

File: src/assets/styles/theme.scss
Comment (argush3) [inline src/assets/styles/theme.scss:71]: i like it. give us a base to work with.  and some sanity too.

Comment (severinbeauvais) [reply src/assets/styles/theme.scss:71]: Thanks!

And visibility, too. And it's much easier to search/reference variables than values :)

File: src/assets/styles/theme.scss
Comment (severinbeauvais) [reply src/assets/styles/theme.scss]: I updated the shared component to use `$app-blue` instead, reimported it here, and deleted this obsolete colour.

File: src/mixins/filing-mixin.ts
Comment (severinbeauvais) [reply src/mixins/filing-mixin.ts:135]: I just checked, and Create UI doesn't yet do anything with dissolution type in the draft filing, so the change above shouldn't break anything.

The next dev to work on this will have to sync up Create UI with Filings UI, however.

File: src/components/EntityInfo.vue
Comment (argush3) [reply src/components/EntityInfo.vue:42]: great tip!

File: src/components/EntityInfo.vue
Comment (argush3) [reply src/components/EntityInfo.vue:137]: on another note, i didn't even know about the `dl`, `dt` and `dd` tags.

**Review by argush3 — APPROVED** [review]
LGTM!  Great work.  Lots of cleanup work that will make everyone's life easier.  Thanks!

**Pull request merged**
